### ISSUE TITLE
Using mLSTM with PackedSequence

### ISSUE THREAD

**Issue by diegoantognini** [FlorianWilhelm/mlstm4reco#1]
Hi,

Thank you to have implementing the mLSTM layer, I couldn't find it on the internet.

I was interested to use it but my batch data are packed via the function ```nn.utils.rnn.pack_padded_sequence(embedded, input_lengths, batch_first=True)``` and thus, mLSTM doesn't not handle. I'm using this because my data have different lengths.

Would you have an idea to do it easily ?

Thank you for your help

**Comment by FlorianWilhelm** [comment]
@Diego999, glad you like it. I think this is not so easy. First of all you could look into the implementation of `forward` in [RNNCellBase](https://pytorch.org/docs/stable/_modules/torch/nn/modules/rnn.html#RNNBase). You would have to do the same kind of extra code path for packed inputs. At the end an RNN backend is called and a boolean `variable_length=is_packed` is passed. From that point we are in C++ land and I guess the you will find the corresponding implementation [here](https://github.com/pytorch/pytorch/blob/v0.4.0/aten/src/ATen/native/cudnn/RNN.cpp). So long story short, it will be a lot of work and complicated. I would rather avoid packed sequences in that case and rather save a mask beforehand that tells you which outputs were padding elements.

**Comment by diegoantognini** [comment]
@FlorianWilhelm Thank you very much for your answer ! To be honest, I thought this was the only way and unfortunately, your answer is saying the same. I guest the best way, as you said, is to let the padding elements and put their embedding to the vector 0 (and fixed during training). Only computation time will be longer but at least, it should work.

Thank you for your help !
